stock_analytic/models/stock.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013 Julius Network Solutions
+# Copyright 2015 Clear Corp
+# Copyright 2016 OpenSynergy Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models, api
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    analytic_account_id = fields.Many2one(
+        string='Analytic Account',
+        comodel_name='account.analytic.account',
+    )
+
+
+class StockQuant(models.Model):
+
+    _inherit = "stock.quant"
+
+    @api.model
+    def _prepare_account_move_line(self, move, qty, cost,
+                                   credit_account_id, debit_account_id,
+                                   context=None):
+        res = super(StockQuant, self)._prepare_account_move_line(
+            move, qty, cost, credit_account_id, debit_account_id,
+            context=context
+        )
+
+        # Add analytic account in debit line
```

Comment:
I was wondering then why were you using `range(0, 2)` won't it affect [both the lines?!](https://github.com/odoo/odoo/blob/9.0/addons/stock_account/stock_account.py#L354)